app/models/rental.rb

```diff
@@ -0,0 +1,26 @@
+class Rental < ApplicationRecord
+  belongs_to :customer
+  belongs_to :movie
+
+  validates :customer_id, presence: true
+  validates :movie_id, presence: true
+
+
+  def rent_movie(customer, movie)
+    checked_out = customer.movies_checked_out_count
+    customer.update(movies_checked_out_count: checked_out +1)
+
+    inventory = movie.available_inventory
+    movie.update(available_inventory: inventory - 1)
+  end
+
+
+  def return_movie(customer, movie)
+    checked_out = customer.movies_checked_out_count
+    customer.update(movies_checked_out_count: checked_out -1)
+
+    inventory = movie.available_inventory
+    movie.update(available_inventory: inventory +1)
+  end
```

Comment:
These two methods are so wonderful and perfect. I want to show them off! This is exactly what I was looking for: You pulled out this logic into the `Rental`, which had logic that it delegated to `customer` and `movie`. Nicely done.